tools/biobb_pytorch/biobb_apply_mdae.xml

```diff
@@ -0,0 +1,91 @@
+<tool id="biobb_pytorch_apply_mdae" name="ApplyMdae" version="@TOOL_VERSION@" profile="22.05">
+    <description>Apply a Molecular Dynamics AutoEncoder (MDAE) PyTorch model.</description>
+    <macros>
+      <token name="@TOOL_VERSION@">4.2.1</token>
+    </macros>
+
+    <requirements>
+      <requirement type="package" version="@TOOL_VERSION@">biobb_pytorch</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+
+      ln -s '$input_data_npy_path' ./input_data_npy_path.$input_data_npy_path.ext &&
+      ln -s '$input_model_pth_path' ./input_model_pth_path.$input_model_pth_path.ext &&
+      #if $config_json:
+        ln -s '$config_json' ./config_json.$config_json.ext &&
+      #end if
+
+      apply_mdae
+
+      #if $config_json:
+        --config ./config_json.$config_json.ext
+      #end if
+
+      --input_data_npy_path ./input_data_npy_path.$input_data_npy_path.ext
+      --input_model_pth_path ./input_model_pth_path.$input_model_pth_path.ext
+      --output_reconstructed_data_npy_path $outname_output_reconstructed_data_npy_path
+      #if $outname_output_latent_space_npy_path:
+        --output_latent_space_npy_path $outname_output_latent_space_npy_path
+      #end if
+      ;
+
+      mv '$outname_output_reconstructed_data_npy_path' '$output_reconstructed_data_npy_path' &&
+      if test -f '$outname_output_latent_space_npy_path'; then mv '$outname_output_latent_space_npy_path' '$output_latent_space_npy_path'; fi;
+
+      ]]>
+    </command>
+
+    <inputs>
+      <param name="input_data_npy_path"  type="data" format="npy"  optional="False" label="Input NPY file" help="Input data file"/>
+      <param name="input_model_pth_path" type="data" format="pth"  optional="False" label="Input PTH file" help="Input model file"/>
+      <param name="outname_output_reconstructed_data_npy_path" type="text" value="myapply_mdae.npy" optional="False" label="Output NPY name" help="Output reconstructed data file"/>
+      <param name="outname_output_latent_space_npy_path" type="text" value="myapply_mdae.npy" optional="True" label="Output NPY name" help="Output reduced dimensionality file"/>
+      <param name="config_json" type="data" format="json" optional="True" label="Configuration file" help="File containing tool settings"/>
+    </inputs>
+
+    <outputs>
+      <data format="npy" name="output_reconstructed_data_npy_path" label="output_reconstructed_data_npy_path" />
+      <data format="npy" name="output_latent_space_npy_path" label="output_latent_space_npy_path" />
```

Comment:
If this output is optional you can use a  `filter` here to only define this output when your boolean is true.